pkg/scheduler/framework/plugins/interpodaffinity/scoring.go

```diff
@@ -107,10 +107,13 @@ func (pl *InterPodAffinity) processExistingPod(
 	// For every hard pod affinity term of <existingPod>, if <pod> matches the term,
 	// increment <p.counts> for every node in the cluster with the same <term.TopologyKey>
 	// value as that of <existingPod>'s node by the constant <args.hardPodAffinityWeight>
-	if pl.args.HardPodAffinityWeight > 0 {
-		for _, term := range existingPod.RequiredAffinityTerms {
-			t := framework.WeightedAffinityTerm{AffinityTerm: term, Weight: pl.args.HardPodAffinityWeight}
-			topoScore.processTerm(&t, incomingPod, existingPodNode, 1)
+	if pl.args.HardPodAffinityWeight > 0 && len(existingPodNode.Labels) != 0 {
+		for _, t := range existingPod.RequiredAffinityTerms {
+			// The existing pod NamespaceSelector is not merged with the Namespaces set, and so
+			// here we explicitly check if there is a match with either.
+			if IncomingPodMatchesTerms(incomingPod, state.namespaceLabels, t.Selector, t.Namespaces, t.NamespaceSelector) {
```

Comment:
Let's change this to `IncomingPodMatchesTerm(incomingPod, state.namespaceLabels, t)`